### ISSUE TITLE
Documentation updates desired for Notation v1.0.0

### ISSUE THREAD

**Issue by dtzar** [notaryproject/notaryproject.dev#77]
Epic Issue to track all the issues we want to complete for the RC-1 release.

- [ ] Add content or remove https://notaryproject.dev/docs/tutorials/
- [ ] Add content or remove https://notaryproject.dev/docs/how-to/
- [ ] Add content or remove https://notaryproject.dev/docs/troubleshooting/
- [x] https://github.com/notaryproject/notaryproject.dev/issues/16

--------
Specifically, The following guides are necessary for the RC.1 release, let's confirm and focus on the RC.1 scope first:

- [x]  [Quick Start](https://notaryproject.dev/docs/quickstart/): update it to be consistent with RC.1
- [ ] [Tutorial](https://notaryproject.dev/docs/tutorials/): user manual for Notation CLI functionalities (basic)
    - [x] #101
    - [x] #102
    - [x] #105
    - [x] #103
    - [x] #108
    - [ ] [Customize the Notation configuration file](https://notaryproject.dev/docs/tutorials/notation-config-file/) (Only check and update if it is inconsistent with RC.1)
    - [ ] [Notation directory structure for system configuration](https://notaryproject.dev/docs/tutorials/directory-structure/) (Only check and update if it is inconsistent with RC.1)
    - [x] #104
- [ ] [How to](https://notaryproject.dev/docs/how-to/): demonstrating end-to-end use cases with ecosystem tools (advanced)
   - [ ] Sign and verify with KMS provider (e.g. Azure, AWS): This will be a page that just lists the related doc links.
   - [ ] #107
- [ ] [Troubleshooting](https://notaryproject.dev/docs/troubleshooting/)
   - [ ] #106

**Comment by FeynmanZhou** [comment]
@dtzar  For the chapters still with empty content, I think we can provide a documentation list first to showcase what user guides need to be added. Then we can start to write the content and ask for technical docs support from the [CNCF service desk portal](https://cncfservicedesk.atlassian.net/servicedesk/customer/portal/1) in parallel.

**Comment by yizha1** [comment]
Here is a proposal for inspiring and  further discussions: @dtzar @FeynmanZhou @iamsamirzon @vaninrao10 

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:dt="uuid:C2F41010-65B3-11d1-A29F-00AA00C14882"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=OneNote.File>
<meta name=Generator content="Microsoft OneNote 15">
</head>

<body lang=en-US style='font-family:Calibri;font-size:11.0pt'>


<div style='direction:ltr'>



Category | Documents   plan | Status | Need   review existing docs? | refreshing
-- | -- | -- | -- | --
Landing page | Redesign a new   landing page | 0 |   | Low frequency
Project Overview | Overview document | 1/1 | Yes | per new release
Quick Start | Quick start document | 1/1 | Yes | per new release
Installation   guides | Installation guide | 1/1 | Yes | per new release
Tutorials | * Sign a container image <br>        * using a local key for testing purpose<br>     * Sign a container image         using a key stored in Key vault (Azure Key vault or AWS KMS)<br>   * Sign artifacts (e.g SBOM file) stored in a registry<br>     * List artifacts and         associated signatures stored in a registry <br>    * Verify signatures using trust store and trust policy<br>    * Build a plugin <br> * Others: like Ratify validation | 0 |   | Per new feature
How-to guides | * Explain Directory         structure after installation of notation CLI <br>    * Configure signing keys <br>    * Configure a different         signature envelope <br>    * Configure trust policy <br>    * Configure trust store <br>    * Install/uninstall plugins | 0 |   | Per new feature
Concepts | Concept | 1/1 | Yes | per new release
Troubleshooting | Troubleshooting guide | 0/1 |   | Per new feature
Frequently asked   questions | FAQ | 1/1 | Yes | per new release
specifications | A page to add links to         each spec in different repos under notaryproject | 0 |   | Per new spec
Announcements | * Release announcements <br>    * Key feature announcement | 0 |   | per new key   feature or per new release
Blog | * Cloud provider story (e.g. AWS or Azure) <br>        * Other types of documents  not in Tutorials and how-to categories           (e.g. Secure supply chain          related）<br> |  0 |   | per new release
Roadmap | https://github.com/notaryproject/roadmap |  1 | Yes | Per milestone   update



</div>


</body>

</html>

**Comment by FeynmanZhou** [comment]
Specifically, I think the following guides are necessary for the RC.1 release, let's confirm and focus on the RC.1 scope first:

- [ ] Update [Quick start](https://notaryproject.dev/docs/quickstart/) to RC.1

- [ ] [Tutorial](https://notaryproject.dev/docs/tutorials/): user manual for Notation CLI functionalities
    - [ ] Sign artifacts
    - [ ] Verify artifacts using the trust store and trust policy
    - [ ] Configure trust store and trust policy
    - [ ] Authentication
    - [ ] Key generation and management
    - [ ] Plugin management
    - [ ] [Customize the Notation configuration file](https://notaryproject.dev/docs/tutorials/notation-config-file/) (Only check and update if it is consistent with RC.1)
    - [ ] [Notation directory structure for system configuration](https://notaryproject.dev/docs/tutorials/directory-structure/) (Only check and update if it is consistent with RC.1)
    - [ ] Registry Support: listed the compatible registries

- [ ] [How to](https://notaryproject.dev/docs/how-to/): demonstrating end-to-end use cases with ecosystem tools
   - [ ] Sign and verify with KMS provider (Azure, AWS): This will be a page that just lists the related doc links.
   - [ ] Sign and verify with Ratify in Kubernetes

@notaryproject/notary-website-maintainer @zr-msft

**Comment by yizha1** [comment]
@FeynmanZhou  I think this is a good list for rc.1. Just questions on the two tutorials below. What is the difference? Did Zach already work on one of the tutorials?

-  Verify artifacts using the trust store and trust policy
-  Configure trust store and trust policy

**Comment by zr-msft** [comment]
agreed with @yizha1  that we can combine configure and verify with trust store and policy

also, i think we should reverse the items under 'tutorials' and 'how-to'. How-to guides are single subject around a feature, tutorials are end-to-end scenarios.

**Comment by vaninrao10** [comment]
> 

I am not sure if we should combine. I would say for a feature which is the tutorial scenario, the per-requisite is what is defined in the How-to guide. If we start putting per-requisites and necessary configuration needed in the tutorial section then it will become massive information for customers. The sectional is good idea.

**Comment by zr-msft** [comment]
> About "Sign and verify with KMS provider (Azure, AWS): This will be a page that just lists the related doc links." a page with links only is not the most helpful documentation. Can this be on a page describing scenarios and how Notary can be used?

@toddysm I'm fine with adding an 'additional scenarios' page that would include this along with some contextual information for each item we're linking out to

@FeynmanZhou  @toddysm I'm assuming we need to include #84 in this list as well

@FeynmanZhou around organization, a lot of the items you have listed under 'tutorials' come across as recipes and not 'leading people by the hand through a scenario'. Regardless, i think we should start focusing on creating the content (one doc per PR) and we can discuss in that PR where it goes. Also, i think each doc needs its own issue to track who is working on it.

**Comment by zr-msft** [comment]
for RC1: tutorial for manually creating a trust policy #98

**Comment by FeynmanZhou** [comment]
> a lot of the items you have listed under 'tutorials' come across as recipes and not 'leading people by the hand through a scenario'. 

@zr-msft Yes. Agree with you. They are just rough recipes for the Notation features but are not the final document title. We can adjust each doc title and content to lead users to resolve a specific scenario.

> i think each doc needs its own issue to track who is working on it.

I will create issues for them accordingly. Then we can claim different issues and work in parallel.

**Comment by zr-msft** [comment]
@toddysm @FeynmanZhou Since this issue refers to an outdated release, we should close or rename. Do either of you have a preference?

**Comment by yizha1** [comment]
@FeynmanZhou  should we close it or update it? And the milestone should be changed accordingly

**Comment by FeynmanZhou** [comment]
I would suggest closing this one and using separate issues to track the remaining documentation items for v1.0.0 since we have finished the discussion and most of the listed items above. Thanks.
